web-console/3-project-details/labels.md

```diff
@@ -0,0 +1,99 @@
+- The save button should not be enabled until selections are made.
+
+![Label 1](img/new-labels-03-add.png)
+
+- The dropdown list should display all existing labels that have been applied elsewhere throughout the current project.
+
+#### Implementation Details
+- Label Names that have already been applied to the current object will not appear in the dropdown list as there cannot be duplicate values associated.
+  - For example, "app" and "openshift.io/deployment-config.name" are already applied and are not in the dropdown list.
+  - Users would need to edit the existing rows for those labels if desired.  
+
+![Label 1](img/new-labels-04-add.png)
+
+![Label 1](img/new-labels-05-add.png)
+
+- Once the user clicks `Save` the new label should appear at the bottom of the list of labels on the Details tab.
+
+
+## Edit Labels
+![Label 1](img/new-labels-01-edit.png)
+
+- The same `Manage Labels` link will be used for editing existing labels.
+- Users can change a name or value as desired. All existing values for the current project will be listed in the dropdown.
+
+#### Implementation Details
+- Values can be edited in this view, but only the current object will receive the label change. In order to bulk change labels for multiple items (Deployments in this case), users must go to the summary page and multi-select items.
+- If there are specific system defined or restricted labels that users are unable to change, the entire row should be disabled.
+
+## Remove Labels
+![Label 1](img/new-labels-01-delete.png)
+
+- The same `Manage Labels` link will be used for removing existing labels.
+- Clicking the trash icon next to a label will remove that row and enable the save button.
+- Applying these changes will only remove the label from this specific object, not from the system entirely.
+
+![Label 1](img/new-labels-02-delete.png)
+
+# Labels on Summary Pages
+
+![Label 1](img/new-labels-01-manage.png)
+
+- Summary lists are provided for each major section such as Deployments, Pods, Builds, Images, Storage, etc.
+- Currently users can filter those lists based on their applied labels, in order to narrow the results.
+- Filtering by label should remain, but checkboxes should be added to the left side of each row to allow for multi-select and bulk actions from the summary page.
+
+## Manage Labels
+- Users can multi-select using the checkboxes provided and a `Manage Labels` button should be available in the top right corner of the page.
+- The `Manage Labels` button will open a modal for any add, edit, or remove actions to take place.
+  - The top of the modal should summarize the selections to reiterate the number of objects that will be affected by the label changes. Intro text detailed below:
+  - "You have selected 'x' 'Object-Type.' The following changes will be applied to all selections. Note: bulk edits can only be made for objects with common labels."
```

Comment:
> Note: bulk edits can only be made for objects with common labels.

Does this mean that you can only change an existing label value if all resources have that label? The wording almost sounds like you can't even add new labels if they don't have common values.

What happens if I try to add a label that is present on one resource and not another? In this case, we could set the new value on all resources. I'm not sure we need to restrict it.

Reply:
I think you'd definitely want to warn the user if adding a label would change/overwrite an existing label.

Reply:
Great points, I am updating the PR to reflect the comments and allow users to overwrite change (after accepting the warning shown).